pinot-common/src/main/java/org/apache/pinot/common/utils/ExceptionUtils.java

```diff
@@ -34,6 +37,21 @@ public static String consolidateExceptionMessages(Throwable e) {
     return sb.toString();
   }
 
+  public static String consolidateExceptionMessages(Throwable e) {
+    StringBuilder sb = new StringBuilder();
+    while (e != null) {
```

Comment:
IIRC there are cases where `e.getCause() = e`. In those cases, this will loop forever. Can we break in that condition here as well?

Reply:
Nice catch (and a smelling situation if that happens). Added a simple check breaking the loop in that case.